tests/lib/args_validator.py

```diff
@@ -257,6 +257,123 @@ def test_ValidateOrder(self):
+    """Test successful validtion."""
+    fqdns = ['github.com',
+             'grr-client-ubuntu.c.ramoj-playground.internal',
+             'www.google.com.au',
+             'www.google.co.uk']
+    for fqdn in fqdns:
+      self.validator.Validate(fqdn)
+
+    self.validator.Validate(','.join(fqdns), {'comma_separated': True})
+
+  def test_ValidationFailure(self):
+    """Tests validation failures."""
+    fqdns = ['value', 'a-.com', '-a.com']
+    for fqdn in fqdns:
+      with self.assertRaisesRegex(
+          errors.RecipeArgsValidatorError,
+          f"'{fqdn}' is an invalid hostname."):
+        self.validator.Validate(fqdn)
+
+    with self.assertRaisesRegex(
+        errors.RecipeArgsValidatorError,
+        "'value' is an invalid hostname."):  # Fails on the first listed
+      self.validator.Validate(','.join(fqdns), {'comma_separated': True})
+
+
+class GRRHostValidatorTest(unittest.TestCase):
+  """Tests the GRRHostValidator class."""
+
+  def setUp(self):
+    """Setup."""
+    self.validator = args_validator.GRRHostValidator()
+
+  def test_Init(self):
+    """Tests initialisation."""
+    self.assertEqual(self.validator.NAME, 'grr_host')
+
+  def test_ValidateSuccess(self):
+    """Test successful validtion."""
+    fqdns = ['C.1facf5562db006ad',
+             'grr-client-ubuntu.c.ramoj-playground.internal']
+    for fqdn in fqdns:
+      self.validator.Validate(fqdn)
+
+    self.validator.Validate(','.join(fqdns), {'comma_separated': True})
+
+  def test_ValidationFailure(self):
+    """Tests validation failures."""
+    fqdns = ['value', 'C.a', 'C.01234567890123456789']
+    for fqdn in fqdns:
+      with self.assertRaisesRegex(
+          errors.RecipeArgsValidatorError,
+          f"'{fqdn}' is an invalid Grr host ID."):
+        self.validator.Validate(fqdn)
+
+    with self.assertRaisesRegex(
+        errors.RecipeArgsValidatorError,
+        "'value' is an invalid Grr host ID."):
+      self.validator.Validate(','.join(fqdns), {'comma_separated': True})
+
+
+class URLValidatorTest(unittest.TestCase):
+  """Tests the URLValidator class."""
+
+  def setUp(self):
+    """Setup."""
+    self.validator = args_validator.URLValidator()
+
+  def test_Init(self):
+    """Tests initialisation."""
+    self.assertEqual(self.validator.NAME, 'url')
+
+  def test_ValidateSuccess(self):
+    """Test successful validation."""
+    fqdns = [
+        'http://10.100.0.100:8080',
+        'http://10.100.0.100',
+        'https://10.100.0.100',
+        'http://grr.ramoj-playground.internal:8080',
+        'http://grr.ramoj-playground.internal',
+        'https://grr.ramoj-playground.internal',
+    ]
+    for fqdn in fqdns:
+      self.validator.Validate(fqdn)
+
+    self.validator.Validate(','.join(fqdns), {'comma_separated': True})
+
+  def test_ValidationFailure(self):
+    """Tests validation failures."""
+    fqdns = [
+        'value',
+        '10.100.0.100',  # Needs scheme
+        'http://one.*.com'
+        ]
```

Comment:
i'd prob remove indentation here or bring it up to the next line

Reply:
Done.